Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -716,16 +732,23 @@ void HexEditor::paint_event(GUI::PaintEvent& event)
 
             painter.fill_rect(text_background_rect, background_color);
             auto character = String::formatted("{:c}", isprint(cell_value) ? cell_value : '.').release_value_but_fixme_should_propagate_errors();
-            painter.draw_text(text_display_rect, character, Gfx::TextAlignment::TopLeft, text_color);
 
             if (m_edit_mode == EditMode::Text) {
                 if (byte_position == m_position) {
                     Gfx::IntRect cursor_position_rect {
-                        text_display_rect.left(), text_display_rect.top(), 2, text_display_rect.height()
+                        text_display_rect.left(),
+                        static_cast<int>(frame_thickness() + m_line_spacing / 2 + i * line_height()),
+                        static_cast<int>(character_width()),
+                        static_cast<int>(line_height())
                     };
-                    painter.fill_rect(cursor_position_rect, palette().text_cursor());
+                    painter.fill_rect(cursor_position_rect, palette().black());
                 }
             }
+
+            if (byte_position == m_position)
+                painter.draw_text(text_display_rect, character, Gfx::TextAlignment::TopLeft, palette().selection_text());
+            else
+                painter.draw_text(text_display_rect, character, Gfx::TextAlignment::TopLeft, text_color);
```

Comment:
This block looks really the one above, what are your thought about factorizing them?

Reply:
I refactored double initialization of background/text colors, you can take a peek in the new commit, IMO looks a little bit clearer but I'm not sure what you think. This code that you've pointed out above has changed a bit since I've addressed your report about a regression, but I've tried to move this chunk in lines 736-746 to a separate lamba as we've discussed on discord, which would remove some repetition, but I don't know why (as also discussed on discord) it seems that when we do that, black box cursor gets drawn on top of the text after this modification. It's weird, since text is being drawn AFTER the call to lambda. I can debug a little bit further if you want, if you'd care to help I'd gladly take your assistance, but I couldn't wrap my head why it happens.

+ also added one more commit since I've found (and fixed) a bug where you could go outside document's boundaries with GoToOffsetDialog.